contracts/asset-proxy/contracts/src/bridges/Eth2DaiBridge.sol

```diff
@@ -0,0 +1,130 @@
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/interfaces/IERC20Token.sol";
+import "@0x/contracts-exchange/contracts/src/interfaces/IWallet.sol";
+import "./ERC20Bridge.sol";
+import "../interfaces/IEth2Dai.sol";
+
+
+// solhint-disable space-after-comma
+contract Eth2DaiBridge is
+    ERC20Bridge,
+    IWallet
+{
+    bytes4 private constant LEGACY_WALLET_MAGIC_VALUE = 0xb0671381;
+    /* Mainnet addresses */
+    address constant public ETH2DAI_ADDRESS = 0x39755357759cE0d7f32dC8dC45414CCa409AE24e;
+    address constant public WETH_ADDRESS = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address constant public DAI_ADDRESS = 0x89d24A6b4CcB1B6fAA2625fE562bDD9a23260359;
+
+    constructor() public {
+        // Grant the Eth2Dai contract unlimited weth and dai allowances.
+        _getWethContract().approve(address(_getEth2DaiContract()), uint256(-1));
+        _getDaiContract().approve(address(_getEth2DaiContract()), uint256(-1));
+    }
+
+    /// @dev Callback for `IERC20Bridge`. Tries to buy `amount` of
+    ///      `toTokenAddress` tokens by selling the entirety of the opposing asset
+    ///      (DAI or WETH) to the Eth2Dai contract, then transfers the bought
+    ///      tokens to `to`.
+    /// @param toTokenAddress The token to give to `to` (either DAI or WETH).
+    /// @param to The recipient of the bought tokens.
+    /// @param amount Minimum amount of `toTokenAddress` tokens to buy.
+    /// @return success The magic bytes if successful.
+    function withdrawTo(
+        address toTokenAddress,
+        address /* from */,
+        address to,
+        uint256 amount,
+        bytes calldata /* bridgeData */
+    )
+        external
+        returns (bytes4 success)
+    {
+        // The "from" token is the opposite of the "to" token.
+        IERC20Token fromToken = _getWethContract();
+        IERC20Token toToken = _getDaiContract();
+        // Swap them if necessary.
+        if (toTokenAddress == address(fromToken)) {
+            (fromToken, toToken) = (toToken, fromToken);
+        } else {
+            require(
+                toTokenAddress == address(toToken),
+                "INVALID_ETH2DAI_TOKEN"
+            );
+        }
+        // Try to sell all of this contract's `fromToken` balance.
+        uint256 boughtAmount = _getEth2DaiContract().sellAllAmount(
+            address(fromToken),
+            fromToken.balanceOf(address(this)),
+            address(toToken),
+            amount
+        );
+        // Transfer the converted `toToken`s to `to`.
+        toToken.transfer(to, boughtAmount);
```

Comment:
We should probably check the return value here, just in case a token isn't 100% ERC20 compliant.